src/core/date-formatter.ts

```diff
@@ -0,0 +1,443 @@
+/**
+ * Date Formatter - Handlebars-based date formatting for Seasons & Stars
+ */
+
+import type { CalendarDate } from './calendar-date';
+import type { SeasonsStarsCalendar } from '../types/calendar';
+
+export class DateFormatter {
+  private calendar: SeasonsStarsCalendar;
+  private templateCache: Map<string, Function> = new Map();
+  private handlebarsAvailable: boolean;
+
+  constructor(calendar: SeasonsStarsCalendar) {
+    this.calendar = calendar;
+    this.handlebarsAvailable = this.checkHandlebarsAvailability();
+
+    if (this.handlebarsAvailable) {
+      this.registerCustomHelpers();
+    }
+  }
+
+  /**
+   * Check if Handlebars is available (for test environments)
+   */
+  private checkHandlebarsAvailability(): boolean {
+    try {
+      return typeof Handlebars !== 'undefined' && typeof Handlebars.compile === 'function';
+    } catch {
+      return false;
+    }
```

Comment:
Handlebars is core to Foundry, it should always be available. We don't need to account for the case where it is not.